### DISCUSSION TITLE
1.20.2 Release

### DISCUSSION THREAD

**Discussion by AterAnimAvis** [neoforged/NeoForge#199] [Announcements]
# NeoForge 20.2, for Minecraft 1.20.2, has been released.

---

Whilst the work is still in progress we’d like to keep you up to date which what’s to come. We have a lot of new technology part as part of this release, which we plan to elaborate further within a future blog post. All of the migration guides will be provided in the blog post as well. Howevever, here are a few things to expect as we release NeoForge.

#### Versioning
NeoForge releases will be using the following version format going forward: `<minecraft_minor>.<minecraft_patch>.<number>(-beta)`. This means that all releases for Minecraft 1.20.2 will be formatted as `20.2.*`. The `-beta` tag indicates that the release is unstable and will likely have breaking changes between 'number' versions.

#### NeoGradle: The Next Generation
Orion (@marchermans) has been working hard on improving the NeoGradle plugin to make all aspects of mod development easier. This means a complete overhaul in the build process, including massive simplifications of your build scripts.

#### MojMaps Everywhere (and a bit of Parchment too)
We are swapping out SRG names in favor of using Mojang's Mapping Set throughout the entire codebase to simplify our toolchain, along with a couple of other benefits. Additionally, development on NeoForge now uses Parchment for parameter and Javadocs.

#### Code changes
All packages names will be migrated from `net.minecraftforge.*` to `net.neoforged.*`. There are also numerous overhauls to systems, like capabilities and registries, that are targetted for the 20.2 release.

We would like to thank @Technici4n, @Shadows-of-Fire, @Matyrobbrt, @SizableShrimp, @Minecraftschurli and the others on the team for all of their hard work!

#### Keeping up to date

If you liked to kept more informed about our current progress make sure to check out our current [Work List for 20.2](https://hackmd.io/@neoforged/SyPzw1Ryp) or to keep up to date and join in on discussion on future changes consider joining our [Discord](https://discord.neoforged.net/).

**Comment by Speiger** [comment]
Small warning regarding the use of "Parchment", forge/neoforge gradle + Parchment crashes/fails if you try to use local file jars.
While the use of maven is ofc a good idea, sometimes it is needed to use local jars, even temporary.

Not sure if this accounted for but it is a good thing to know.

Edit: it seems that issue is solved with 1.20.2 neoforge. Have to validate it but we will see ^^

**Reply by lukebemish** [reply]
`implementation fg.deobf(project.files('whatever.jar'))` works fine for me with parchment and neogradle; not sure what you're doing. Is there an open issue on the librarian repository for this?

**Reply by AterAnimAvis** [reply]
I was unaware that raw files worked, last I knew (Forge/Neo)Gradle did have issues with deobfuscating non maven provided dependencies, although I don't know how NeoGradle-Next handles them.

The recommended way, last I knew, was to use a [`flatDir` maven repository](https://forge.gemwire.uk/wiki/Dependencies#Flat_Directory_Repositories) to provide them.

**Reply by Speiger** [reply]
I have experienced with others some crashes that worked fine with mojang mappings, but crashed with Parchment mappings.
The error was: "Could not find :myMinecraftLibrary:1.0.0_mapped_parchment_2022.11.27-1.19.2."
```groovy
repositories {

	flatDir {
		dirs("library")
	}
}

dependencies {
	implementation fg.deobf(":myMinecraftLibrary:1.0.0")
}
```
This would be the error/code example that we ran into.

And this is the usual suggested way of doing things when you google around.

edit 
@lukebemish there is no official issue of this because of it being a private repo/pms.

@AterAnimAvis forge gradle can handle local file libraries well, though the one thing forge gradle can not handle well if 1 library has another library as JIJ and you try to import a newer version via JIJ next to the library that uses it as Library.

- MyMod
    - Library Mod 1
       - JavaLibrary Version 0.9
    - JavaLibrary Version 1.0

This would fail catastrophically where gradle would refused to compile...

Edit2: I look into making that a official issue, but that might take some time since i am not the person specifically having these issues.
The question is where would i report that?
NeoGradle? Or NeoForge?

**Reply by Minecraftschurli** [reply]
this will be a non issue in `20.2` since there will be no `fg.deobf` anymore messing it up. additionally you will be able to just add mods to your mods folder without needing to add them to your `build.gradle` if you only need them at runtime and not to compile against.

**Reply by Speiger** [reply]
@Minecraftschurli that doesn't solve the issue of requiring sourcecode access in development space.
So in other words you have to decompile mods manually and then drag/drop the src into your project is what you are saying?

**Reply by Technici4n** [reply]
if you need source access it is `implementation <standard dependency without fg.deobf>`, if you just need it at runtime to test something the mod jar goes into `run/mods/`

**Reply by Speiger** [reply]
@Technici4n does that decompile mods too? In other words i can hook any mod up even cursemaven ones or downloaded ones?

**Reply by Minecraftschurli** [reply]
mods don't get obfuscated anymore in 20.2 so it just works as a dependency
